apps/coordinator/ts/redis/__tests__/redis.service.test.ts

```diff
@@ -0,0 +1,122 @@
+
+const REDIS__GET_ALL_PREFIX = "*-test";
+
+const scheduledPoll: IScheduledPoll = {
+  maciAddress: "0xb83074Ac11fc569AC12F1b7D0C0a6809c3dc355b",
+  pollId: "5",
+  mode: 1,
+  chain: ESupportedNetworks.OPTIMISM_SEPOLIA,
+  endDate: 1752534000,
+};
+
+describe("RedisService", () => {
+  let service: RedisService;
+
+  beforeAll(async () => {
+    service = new RedisService();
+    await service.onModuleInit();
+    // Clean up before tests
+    const keys = await service.client.keys(REDIS__GET_ALL_PREFIX);
+    if (keys.length) {
+      await service.client.del(keys);
+    }
+  });
+
+  afterAll(async () => {
+    // Clean up after tests
+    const keys = await service.client.keys(REDIS__GET_ALL_PREFIX);
+    if (keys.length) {
+      await service.client.del(keys);
+    }
+
+    await service.client.quit();
+  });
+
+  it("should connect to Redis without issues", () => {
+    expect(service.client.isOpen).toBe(true);
+  });
+
+  it("should set and get a value, and parse it as IScheduledPoll", async () => {
+    const key = getPollKeyFromObject(scheduledPoll, true);
+    const value = JSON.stringify(scheduledPoll);
+
+    await service.set(key, value);
+
+    const stored = await service.get(key);
+    expect(stored).not.toBeNull();
+
+    const parsed = JSON.parse(stored!) as IScheduledPoll;
+
+    expect(parsed).toEqual(scheduledPoll);
+  });
+
+  it("should return null when non-existent key is requested", async () => {
+    const nonExistentKey = "non-existent-key";
+    const result = await service.get(nonExistentKey);
+
+    expect(result).toBeNull();
+  });
+
+  it("should set and get multiple values", async () => {
+    const pollOne = JSON.parse(JSON.stringify(scheduledPoll)) as IScheduledPoll;
+    pollOne.pollId = "1";
+    const pollKeyOne = getPollKeyFromObject(pollOne, true);
+    await service.set(pollKeyOne, JSON.stringify(pollOne));
+
+    const pollTwo = JSON.parse(JSON.stringify(scheduledPoll)) as IScheduledPoll;
+    pollTwo.pollId = "2";
+    const pollKeyTwo = getPollKeyFromObject(pollTwo, true);
+    await service.set(pollKeyTwo, JSON.stringify(pollTwo));
+
+    const pollThree = JSON.parse(JSON.stringify(scheduledPoll)) as IScheduledPoll;
+    pollThree.pollId = "3";
+    const pollKeyThree = getPollKeyFromObject(pollThree, true);
+    await service.set(pollKeyThree, JSON.stringify(pollThree));
+
+    const storedPollOne = JSON.parse((await service.get(pollKeyOne)) || "{}") as IScheduledPoll;
+    const storedPollTwo = JSON.parse((await service.get(pollKeyTwo)) || "{}") as IScheduledPoll;
+    const storedPollThree = JSON.parse((await service.get(pollKeyThree)) || "{}") as IScheduledPoll;
+
+    expect(pollOne).toEqual(storedPollOne);
+    expect(pollTwo).toEqual(storedPollTwo);
+    expect(pollThree).toEqual(storedPollThree);
+  });
+
+  it("should retrieve all values using getAll", async () => {
+    // 1 entry added in the set and get value test
+    // 3 entries added in the set and get multiple values test
```

Comment:
Not too opinionated but what do you think about writing isolated unit tests which do not rely on state from previous tests?

It could cause unexpected failures if other tests would change for example. For this sort of logic where we want to test sequential state change logic, we could write an integration test/dedicated unit test depending on how many components it involved